src/game/client/swarm/asw_in_mouse.cpp

```diff
@@ -85,6 +88,36 @@ void CASWInput::ApplyMouse( int nSlot, QAngle& viewangles, CUserCmd *cmd, float
 	int current_posx, current_posy;
 	GetMousePos(current_posx, current_posy);
 
+	// restrict cursor to 16:9 area to prevent ultra-wide fov cheat
+	g_ultra_wide_screen = false;
+	C_BasePlayer* player = C_BasePlayer::GetLocalPlayer();
+
+	if (player && !player->IsObserver()) {
```

Comment:
spectating is determined in AS:RD by whether the player has a non-null SpectatingNPC